crates/rolldown/tests/esbuild/default/export_forms_common_js/artifacts.snap

```diff
@@ -18,7 +18,7 @@ var init_a = __esm({ "a.js"() {
 //#endregion
 //#region b.js
 var b_exports = {};
-__export(b_exports, { xyz: () => xyz });
+__export(b_exports, { xyz });
```

Comment:
@IWANABETHATGUY Do you think we could move `namespace decl` down to then bottom so this won't get the error of `reference value before declaring`?

This will surely breaks the esbuild diff I guess.

Reply:
Do you see any discussion that why esbuild don't do this optimization? I am curious why they don't have this optimization.

Reply:
No. I didn't find related discussions. I guess it's just another optimization that esbuild didn't do.

But rollup did this https://rollupjs.org/repl/?version=4.17.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwZm9vJTIwZnJvbSUyMCcuJTJGZm9vLmpzJyU1Q24lNUNuY29uc29sZS5sb2coZm9vKSUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBhJTIwJTNEJTIwMSU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMHMlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyZm9vLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMm91dHB1dCUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmlpZmUlMjIlMkMlMjJuYW1lJTIyJTNBJTIyQSUyMiU3RCU3RCU3RA==.

Reply:
It requires some analysis to do this optimization, I guess they didn't do it due to complexity and not have manpower  to put effort on this. And we have semantic analysis from oxc. So they save the time for semantic analysis.

Reply:
I think it's fine to put declaration before `__export`

Reply:
But I hope it could be done first in another pr, so it will not introduce too much noise.

Reply:
Yeah. it'll be done in another PR. I'm fighting graphite.